include/wil/wistd_type_traits.h

```diff
@@ -35,6 +35,13 @@
 #ifndef _WISTD_TYPE_TRAITS_H_
 #define _WISTD_TYPE_TRAITS_H_
 
+#ifdef __MINGW32__
+#include <type_traits>
+#define WI_IS_CONVERTIBLE_TO(A, B) std::is_convertible<A, B>::value
+#else
+#define WI_IS_CONVERTIBLE_TO(A, B) __is_convertible_to(A, B)
+#endif

```

Comment:
Looks like the original usage of `__is_convertible_to` is guarded by `__WI_HAS_FEATURE_IS_CONVERTIBLE_TO`. I'd expect that `__WI_HAS_FEATURE_IS_CONVERTIBLE_TO` would be 0 if `__is_convertible_to` is not available (unless GCC uses a different spelling?). If there is a conflict, perhaps we can just force `__WI_HAS_FEATURE_IS_CONVERTIBLE_TO` to 0 for mingw?